src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService_Helpers.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.InheritanceMargin
+{
+    internal partial class AbstractInheritanceMarginService
+    {
+        private static async Task<InheritanceMemberItem> CreateInheritanceMemberInfoAsync(
+            Solution solution,
+            INamedTypeSymbol memberSymbol,
+            int lineNumber,
+            ImmutableArray<ISymbol> baseSymbols,
+            ImmutableArray<ISymbol> derivedTypesSymbols,
+            CancellationToken cancellationToken)
+        {
+            var baseSymbolItems = await baseSymbols
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Implementing, cancellationToken))
+                .ConfigureAwait(false);
+
+            var derivedTypeItems = await derivedTypesSymbols
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Implemented, cancellationToken))
+                .ConfigureAwait(false);
+
+            return new InheritanceMemberItem(
+                lineNumber,
+                memberSymbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                memberSymbol.GetGlyph(),
+                baseSymbolItems.Concat(derivedTypeItems));
+        }
+
+        private static async ValueTask<InheritanceTargetItem> CreateInheritanceItemAsync(
+            Solution solution,
+            ISymbol targetSymbol,
+            InheritanceRelationship inheritanceRelationshipWithOriginalMember,
```

Comment:
```suggestion
            InheritanceRelationship inheritanceRelationship,
```

i think the 'withoriginalmember' is properly implied here.